include/dsn/internal/task.h

```diff
@@ -118,6 +120,7 @@ class task :
     bool                    cancel(bool wait_until_finished, /*out*/ bool* finished = nullptr);
     bool                    wait(int timeout_milliseconds = TIME_MS_MAX, bool on_cancel = false);
     virtual void            enqueue();
+    bool                    set_retry(bool enqueue_immediately = true); // return true when called inside exec(), false otherwise
```

Comment:
Will improve in later pr. On the other hand, the state check is the same for the "called inside exec()", as when in exec(), the state is guaranteed to be RUNNING, otherwise we have a bug.